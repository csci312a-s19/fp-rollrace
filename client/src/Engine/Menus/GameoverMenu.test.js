import React from 'react';
import GameOverMenu from './GameoverMenu.js';
import { shallow, mount } from 'enzyme';

describe('Game over menu tests', () => {
  const score = 0;
  const restart = jest.fn();
  const exitToMenu = jest.fn();
  const showModal = true;
  let comp;

  //snapshot for reference
  describe('Change Key menu snapshot test', () => {
    test('progressBar snapshot', () => {
      comp = shallow(
        <GameOverMenu
          score={score}
          restart={restart}
          exitToMenu={exitToMenu}
          showModal={showModal}
        />
      );
      expect(comp).toMatchSnapshot();
    });
  });

  describe('Game over menu functionality tests', () => {
    beforeEach(() => {
      comp = mount(
        <GameOverMenu
          score={score}
          restart={restart}
          exitToMenu={exitToMenu}
          showModal={showModal}
        />
      );
    });

    test('Props testing', () => {
      expect(comp.props().score).toEqual(0);
      expect(comp.props().showModal).toEqual(true);
    });

    test('Model Components are in place', () => {
      expect(comp.find('.modalprovider').exists()).toEqual(true);
      expect(comp.find('.styledmodal').exists()).toEqual(true);
      expect(comp.find('.prompt').exists()).toEqual(true);
      expect(comp.find('.score').exists()).toEqual(true);
      expect(comp.find('.highscore').exists()).toEqual(true);
    });

    test('Modal pause menu contains necessary buttons', () => {
      expect(comp.find('.restart').exists()).toEqual(true);
      expect(comp.find('.goToMenu').exists()).toEqual(true);
    });

    test('Clicking restart button calls correct callback function', () => {
      const restartButton = comp
        .find('.restart')
        .filterWhere(n => n.name() === 'button');
      restartButton.simulate('click');
      expect(restart).toHaveBeenCalledTimes(1);
    });

    test('Clicking main menu button calls correct callback function', () => {
      const mainMenuButton = comp
        .find('.goToMenu')
        .filterWhere(n => n.name() === 'button');
      mainMenuButton.simulate('click');
      expect(exitToMenu).toHaveBeenCalledTimes(1);
    });
  });
});
