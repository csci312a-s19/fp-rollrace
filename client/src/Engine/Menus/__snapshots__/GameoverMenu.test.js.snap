// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game over menu tests Change Key menu snapshot test progressBar snapshot 1`] = `
<t
  className="modalprovider"
>
  <i
    className="styledmodal"
    isOpen={true}
  >
    <styled.p
      className="prompt"
    >
      Game Over!
    </styled.p>
    <styled.p
      className="score"
    >
      Score: 0 sec
    </styled.p>
    <styled.p
      className="highscore"
    >
      Highscore: undefined sec
    </styled.p>
    <styled.button
      className="restart"
      onClick={[MockFunction]}
    >
      Restart
    </styled.button>
    <styled.button
      className="goToMenu"
      onClick={[MockFunction]}
    >
      Main Menu
    </styled.button>
  </i>
</t>
`;
